spec/tree-view-package-spec.coffee

```diff
@@ -2747,391 +2747,393 @@ describe "TreeView", ->
+              expect(notification.getMessage()).toContain 'The following file couldn\'t be moved to the trash'
+              expect(notification.getDetail()).toContain 'test-file.txt'
+
+        it "does nothing when no file is selected", ->
+          atom.notifications.clear()
 
-        treeView.focus()
-        treeView.deselect()
-        atom.commands.dispatch(treeView.element, 'tree-view:remove')
+          spyOn(atom, 'confirm')
 
-        expect(atom.confirm.mostRecentCall).not.toExist()
-        expect(atom.notifications.getNotifications().length).toBe 0
+          treeView.focus()
+          treeView.deselect()
+          atom.commands.dispatch(treeView.element, "tree-view:#{cmd}")
 
-      describe "when a directory is removed", ->
-        it "closes editors with filepaths belonging to the removed folder", ->
-          callback = jasmine.createSpy('onEntryDeleted')
-          treeView.onEntryDeleted(callback)
+          expect(atom.confirm.mostRecentCall).not.toExist()
+          expect(atom.notifications.getNotifications().length).toBe 0
 
-          waitForWorkspaceOpenEvent ->
-            atom.workspace.open(filePath2)
+        describe "when a directory is removed", ->
+          it "closes editors with filepaths belonging to the removed folder", ->
+            callback = jasmine.createSpy('onEntryDeleted')
+            treeView.onEntryDeleted(callback)
 
-          waitForWorkspaceOpenEvent ->
-            atom.workspace.open(filePath3)
+            waitForWorkspaceOpenEvent ->
+              atom.workspace.open(filePath2)
 
-          runs ->
-            openFilePaths = atom.workspace.getTextEditors().map((editor) -> editor.getPath())
-            expect(openFilePaths).toEqual([filePath2, filePath3])
-            dirView2.dispatchEvent(new MouseEvent('click', {bubbles: true, detail: 1}))
-            treeView.focus()
+            waitForWorkspaceOpenEvent ->
+              atom.workspace.open(filePath3)
 
-            spyOn(atom, 'confirm').andCallFake (options, callback) -> callback(0)
+            runs ->
+              openFilePaths = atom.workspace.getTextEditors().map((editor) -> editor.getPath())
+              expect(openFilePaths).toEqual([filePath2, filePath3])
+              dirView2.dispatchEvent(new MouseEvent('click', {bubbles: true, detail: 1}))
+              treeView.focus()
 
-            atom.commands.dispatch(treeView.element, 'tree-view:remove')
+              spyOn(atom, 'confirm').andCallFake (options, callback) -> callback(0)
 
-          waitsFor 'directory to be deleted', ->
-            callback.mostRecentCall.args[0].pathToDelete is dirPath2
+              atom.commands.dispatch(treeView.element, "tree-view:#{cmd}")
 
-          runs ->
-            openFilePaths = atom.workspace.getTextEditors().map((editor) -> editor.getPath())
-            expect(openFilePaths).toEqual([])
+            waitsFor 'directory to be deleted', ->
```

Comment:
I am not so familiar with this old style of waiting for promises. It seems it doesn't actually wait. 

@uzitech Do you know how to make it wait for async functions?

Reply:
I think you have to use `waitsForPromise`

Reply:
Maybe adding to the condition can fix the issue too. Because the promise is not accessible through the test.

```coffee
			  callback.mostRecentCall isnt undefined and
              callback.mostRecentCall.args isnt undefined and
```

Reply:
It seems that the callback isn't called at all.

The first log is from the actual function and the second is what the spy sees.
![image](https://user-images.githubusercontent.com/16418197/114098608-34df7a00-9887-11eb-8534-28df781a9cf1.png)

Reply:
Can't you just copy the tests for `tree-view:remove`?

Reply:
I did, but it doesn't work for some reason. The spies are not called for unknown reasons.